File: Sean的leetcode/Leetcode/Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (!nums.size()) return 0;

        int k = 0;
        int count = 1;
        for (int i = 1; i < nums.size(); i++)
        {
            if (nums[i] != nums[k])
            {
                nums[++k] = nums[i];
                count = 1;
            }
            else
            {
                if (count == 1)
                {
                    nums[++k] = nums[i];
                    count = 2;
                }
            }
        }
        return k+1;
    }
};

File: Sean的leetcode/FB/Remove Duplicates from Sorted Array II.py
###
# Time Complexity: O(n) 
# Space Complexity: O(1)
###
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        l = 0
        for i, num in enumerate(nums):
            if i < 2 or num > nums[l-2]:
                nums[l] = num
                l += 1
        return l